flare/otf.py

```diff
@@ -265,27 +271,36 @@ def run(self):
                     update_threshold=self.update_threshold,
                 )
 
-                if (not std_in_bound) and (
-                    self.steps_since_dft > self.min_steps_with_model
-                ):
+                steps_since_dft = self.curr_step - self.last_dft_step
+                if (not std_in_bound) and (steps_since_dft > self.min_steps_with_model):
                     # record GP forces
                     self.update_temperature()
                     self.record_state()
-                    gp_frcs = deepcopy(self.structure.forces)
+                    gp_energy = self.structure.potential_energy
+                    gp_forces = deepcopy(self.structure.forces)
+                    gp_stress = deepcopy(self.structure.stress)
 
                     # run DFT and record forces
                     self.dft_step = True
-                    self.steps_since_dft = 0
+                    self.last_dft_step = self.curr_step
                     self.run_dft()
                     dft_frcs = deepcopy(self.structure.forces)
                     dft_stress = deepcopy(self.structure.stress)
                     dft_energy = self.structure.potential_energy
 
                     # run MD step & record the state
                     self.record_state()
+                    self.record_dft_data(self.structure, target_atoms)
```

Comment:
A more detailed comment here would be helpful -- e.g. where is the xyz file being stored?

Reply:
added in commit f3c0b55